lib/config/config-ops.js

```diff
@@ -370,5 +370,33 @@ module.exports = {
 
         return patternList.some(pattern => minimatch(filePath, pattern, opts)) &&
             !excludedPatternList.some(excludedPattern => minimatch(filePath, excludedPattern, opts));
+    },
+
+    /**
+     * Normalizes a value for a global in a config
+     * @param {(boolean|string|null)} configuredValue The value given for a global in configuration or in
+     * a global directive comment
+     * @returns {("readable"|"writeable"|"off")} The value normalized as a string
+     */
+    normalizeConfigGlobal(configuredValue) {
+        switch (configuredValue) {
+            case "off":
+                return "off";
+
+            case true:
+            case "true":
+            case "writeable":
+                return "writeable";
+
+            case null:
+            case false:
+            case "false":
+            case "readable":
+                return "readable";
+
+            // Fallback to minimize compatibility impact
+            default:
+                return "writeable";
```

Comment:
Variable writability is communicated to rules via the `variable.writeable` boolean property in scope analysis.

Previously, a non-boolean value in a `globals` configuration would actually get passed directly to rules through scope analysis, so a configuration of `{globals: {myGlobal: "foo"}}` would end up as a variable object with `variable.writeable = "foo"`. 

Strictly speaking, we don't have any compatibility guarantees for non-boolean values here, since it's undocumented behavior. But since someone is probably relying on this by mistake (e.g. from a malformed config comment), I put in a fallback so that unexpected `globals` values default to `"writeable"` (and appear to rules as `writeable: true`). This fallback seems most likely to match existing behavior when users have malformed configs, since rules using this would typically be doing truthiness checks.